src/python/pants/engine/fs.py

```diff
@@ -163,13 +164,32 @@ class DirectoryWithPrefixToStrip(datatype([('directory_digest', Digest), ('prefi
 
 class DirectoryToMaterialize(datatype([('path', str), ('directory_digest', Digest)])):
```

Comment:
Unlike with the v1 version of this call, I think that the `path` in this case should be effectively hardcoded as relative to the buildroot.